src/lib/database.ts

```diff
@@ -68,4 +68,36 @@ export class Database {
       completedMissions: updatedMissions,
     });
   }
+
+  /**
+   * Returns true if able to connect to the Firestore instance.
+   * The Firestore API times out a request after 60 seconds. This method
+   * implements a configurable override that defaults to 5 seconds, but there's
+   * no point in setting it higher than 60 seconds.
+   * @param timeout seconds
+   */
+  async isConnected(timeout: number = 5): Promise<boolean> {
+    try {
+      timeout = Math.min(timeout, 60) * 1000;
+      // eslint-disable-next-line no-undef
+      let timerId: NodeJS.Timeout;
+
+      const timer = new Promise<boolean>((resolve) => {
```

Comment:
If you are keeping `setTimeout`, could you add a unit test or 2 to ensure this is working as expected would be nice. Make sure edgecases are handled.

Reply:
Not sure I understand. No need to poll; if `listCollections` resolves first, it "wins" the race and the function returns.

Reply:
Or sure file one 👍🏼 Also I know I'm coming in from left-field a bit - ty for the explanation 🎊